### PR TITLE
[Enhancement] Support registering function

### PR DESCRIPTION
**Pull request by 274869388**
## Motivation

The motivation of this PR is to support register function.

## Modification

- Registry  
Modify the method `_register_module` to support  register function.

## BC-breaking (Optional)
No

## Use cases (Optional)
```python
from mmcv.cnn import MODELS

@MODELS.register_module()
def build_resnet50():
    pass
```

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by imabackstabber** [comment]
Maybe should correct the test function too.see Line 92 in ``tests/test_utils/test_registry.py``

File: tests/test_utils/test_registry.py [hunk 1]
```diff
@@ -89,12 +89,16 @@ class SphynxCat:
...
-        @CATS.register_module()
-        def some_method():
-            pass
+        class Demo:
+
+            def some_method(self):
+                pass
+
+        method = Demo().some_method
+        CATS.register_module(name='some_method', module=method)
```
Comment (zhouzaida) [inline tests/test_utils/test_registry.py:101]:
```suggestion
        def func1(self):
            pass

        CATS.register_module(name='func1', module=func1)
        
        @CATS.register_module()
        def func2(self):
            pass
```

File: mmcv/utils/registry.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def build_from_cfg(cfg, registry, default_args=None):
         if obj_cls is None:
             raise KeyError(
                 f'{obj_type} is not in the {registry.name} registry')
-    elif inspect.isclass(obj_type):
+    elif inspect.isclass(obj_type) or inspect.isfunction(obj_type):
```
Comment (274869388) [inline mmcv/utils/registry.py:46]: Do we need to add build_func to distinguish between class and function calls?

File: docs/en/understand_mmcv/registry.md [hunk 3]
```diff
@@ -3,11 +3,15 @@
 MMCV implements [registry](https://github.com/open-mmlab/mmcv/blob/master/mmcv/utils/registry.py) to manage different modules that share similar functionalities, e.g., backbones, head, and necks, in detectors.
 Most projects in OpenMMLab use registry to manage modules of datasets and models, such as [MMDetection](https://github.com/open-mmlab/mmdetection), [MMDetection3D](https://github.com/open-mmlab/mmdetection3d), [MMClassification](https://github.com/open-mmlab/mmclassification), [MMEditing](https://github.com/open-mmlab/mmediting), etc.
 
+```{note}
+In v1.4.9 and later, the Registry supports registering functions and calling them.
```
Comment (ZwwWayne) [inline docs/en/understand_mmcv/registry.md:7]: should be 1.5.0 now

**Review by ZwwWayne — APPROVED** [review]

File: docs/en/understand_mmcv/registry.md (hunk 3 above)
Comment (ZwwWayne) [reply docs/en/understand_mmcv/registry.md:7]: 1.5.1

Comment (274869388) [reply docs/en/understand_mmcv/registry.md]: done

File: docs/en/understand_mmcv/registry.md [hunk 4]
```diff
@@ -61,8 +76,10 @@ So you need to import that file somewhere. More details can be found at https://
 If the module is successfully registered, you can use this converter through configs as
 
 ```python
-converter_cfg = dict(type='Converter1', a=a_value, b=b_value)
-converter = CONVERTERS.build(converter_cfg)
+converter1_cfg = dict(type='Converter1', a=a_value, b=b_value)
+converter2_cfg = dict(type='converter2', a=a_value, b=b_value)
+converter1 = CONVERTERS.build(converter1_cfg)
+converter2 = CONVERTERS.build(converter2_cfg)
```
Comment (zhouzaida) [inline docs/en/understand_mmcv/registry.md:82]:
```suggestion
# returns the calling result
result = CONVERTERS.build(converter2_cfg)
```

File: docs/zh_cn/understand_mmcv/registry.md [hunk 5]
```diff
@@ -41,21 +46,34 @@ class Converter1(object):
...
 ```
 如果模块被成功注册了，你可以通过配置文件使用这个转换器（converter），如下所示：
 
 ```python
-converter_cfg = dict(type='Converter1', a=a_value, b=b_value)
-converter = CONVERTERS.build(converter_cfg)
+converter1_cfg = dict(type='Converter1', a=a_value, b=b_value)
+converter2_cfg = dict(type='converter2', a=a_value, b=b_value)
+converter1 = CONVERTERS.build(converter1_cfg)
+converter2 = CONVERTERS.build(converter2_cfg)
```
Comment (zhouzaida) [inline docs/zh_cn/understand_mmcv/registry.md:76]:
```suggestion
# returns the calling result
result = CONVERTERS.build(converter2_cfg)
```

File: mmcv/utils/registry.py [hunk 6]
```diff
@@ -7,7 +7,21 @@
...
+        >>> @MODELS.register_module()
+        >>> class ResNet:
+        >>>     pass
+        >>> resnet = build_from_cfg(dict(type='Resnet'), MODELS)
+        >>> # Returns an instantiated object
+        >>> @MODELS.register_module()
+        >>> def resnet50():
+        >>>     pass
+        >>> resnet = build_from_cfg(dict(type='resnet50'), MODELS)
+        >>> # Return a function call result
```
Comment (zhouzaida) [inline mmcv/utils/registry.py:24]: needs to refine here

Comment (274869388) [reply mmcv/utils/registry.py]: done

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**